openmls/src/group/tests/test_framing_validation.rs

```diff
@@ -325,22 +326,20 @@ fn test_valsem004(ciphersuite: &'static Ciphersuite, backend: &impl OpenMlsCrypt
 
     let original_message = plaintext.clone();
 
-    plaintext.set_sender(Sender {
-        sender_type: SenderType::Member,
-        sender: 100u32,
-    });
+    let random_sender = Sender::build_member(&KeyPackageRef::from_slice(
+        &backend.rand().random_vec(16).unwrap(),
+    ));
+    plaintext.set_sender(random_sender);
 
     let message_in = MlsMessageIn::from(plaintext);
 
     let err = alice_group
         .parse_message(message_in, backend)
-        .expect_err("Could parse message despite wrong sender index.");
+        .expect_err("Could parse message despite wrong sender.");
 
     assert_eq!(
         err,
-        MlsGroupError::Group(CoreGroupError::FramingValidationError(
-            FramingValidationError::UnknownMember
-        ))
+        MlsGroupError::Group(CoreGroupError::TreeSyncError(TreeSyncError::LeafNotInTree))
```

Comment:
Yep, because we have to convert the key package reference into an index first we run into the issue that the leaf doesn't exist a lot before we get to some other check with a more useful message.